server/src/main/java/org/apache/druid/metadata/SQLMetadataStorageActionHandler.java

```diff
@@ -311,7 +322,104 @@ public List<TaskInfo<EntryType, StatusType>> getTaskInfos(
     );
   }
 
-  protected Query<Map<String, Object>> createCompletedTaskInfoQuery(
+  @Override
+  public List<TaskStatusPlus> getTaskStatusPlusList(
+      Map<TaskLookupType, TaskLookup> taskLookups,
+      @Nullable String dataSource,
+      boolean fetchPayload
```

Comment:
This boolean here about fetching payload is weird, especially on a public method.  This class should already know if it shoudl be using the payload or not.  Once you move the mgiration logic into this class, you won't need this anymore.

Reply:
Done